### ISSUE TITLE
java.lang.NullPointerException: Cannot invoke "java.net.InterfaceAddress.getBroadcast()" because "address" is null

### ISSUE THREAD

**Issue by alstergee** [Deep-Symmetry/beat-link#67]
Hi 

so whenever i launch DSBLT it throws this error:

java.lang.NullPointerException: Cannot invoke "java.net.InterfaceAddress.getBroadcast()" because "address" is null
 

full log:

2024-Feb-16 21:41:07.940 FLANKER-RESOLUME INFO [beat-link-trigger.logs:62] - Beat Link Trigger version 7.3.0 built Fri Nov 24 20:54:04 UTC 2023
2024-Feb-16 21:41:07.958 FLANKER-RESOLUME INFO [beat-link-trigger.logs:63] - Java version 17.0.9, OpenJDK 64-Bit Server VM, Amazon.com Inc.
2024-Feb-16 21:41:07.959 FLANKER-RESOLUME INFO [beat-link-trigger.logs:64] - Operating system version Windows 10 10.0
2024-Feb-16 21:41:07.960 FLANKER-RESOLUME INFO [beat-link-trigger.logs:65] - Log files can grow to 200000 bytes, with 5 backlog files.
2024-Feb-16 21:41:07.961 FLANKER-RESOLUME INFO [beat-link-trigger.core:195] - Beat Link Trigger starting.
2024-Feb-16 21:41:08.747 FLANKER-RESOLUME INFO [beat-link-trigger.core:114] - Trying to go online, Use Real Player Number? true
2024-Feb-16 21:41:08.988 FLANKER-RESOLUME WARN [beat-link-trigger.core:152] - Unable to create Virtual CDJ
                  beat_link_trigger.BeatLinkTrigger.main                           
                                                     ...                           
                 beat-link-trigger.BeatLinkTrigger/-main  BeatLinkTrigger.clj:   59
                 beat-link-trigger.BeatLinkTrigger/-main  BeatLinkTrigger.clj:   84
                                                     ...                           
                            beat-link-trigger.core/start             core.clj:  272
                 beat-link-trigger.core/try-going-online             core.clj:  150
              beat-link-trigger.core/try-going-online/fn             core.clj:  150
              org.deepsymmetry.beatlink.VirtualCdj.start      VirtualCdj.java: 1042
   org.deepsymmetry.beatlink.VirtualCdj.createVirtualCdj      VirtualCdj.java:  873
org.deepsymmetry.beatlink.VirtualCdj.findMatchingAddress      VirtualCdj.java:  542
java.lang.NullPointerException: Cannot invoke "java.net.InterfaceAddress.getBroadcast()" because "address" is null

2024-Feb-16 21:41:10.526 FLANKER-RESOLUME WARN [beat-link-trigger.core:178] - Unable to create Virtual CDJ

i know about the issue of anydesk application competing for ports so ive made damn sure thats not running, i also made sure rekordbox and prodjlink arent up at all and disabled firewall / AV just to be sure 

i also uninstalled, found all the folders and registry items left behind and deleted those out, rebooted and reinstalled and no luck.

anyone have any ideas?

**Comment by brunchboy** [comment]
This is unbelievably frustrating but I have to go to bed. I added code to Beat Link to protect against this problem, and tried kicking off a build of Beat Link Trigger using that new version. And for reasons seemingly completely unrelated to any of that, the Github Actions workflow that tries to build the Windows installer is suddenly failing inexplicably. I am going to have to ask the colleague who originally created that script (and used to let me run it on his own server) to take a look and see if he has any insight into it, but I can’t predict when I will next be able to release Windows installers. I was hoping you might be willing to test again with tailscale once this workaround is in, but I can’t predict when that might be now.

**Comment by brunchboy** [comment]
My build is still broken, but I was able to semi-manually create a Windows installer with the proposed fix for you. If you are willing to try that preview release, you can find it here: https://github.com/Deep-Symmetry/beat-link-trigger/releases/tag/v7.4.0-SNAPSHOT

**Comment by alstergee** [comment]
I am long gone from the "office" but when I get back this Wednesday I'm
down to try it out, it would be nice to be able to use tailscale and BLT in
the future. Many thanks for your diligence on this

**Comment by brunchboy** [comment]
No worries, there is no rush! Whenever you have time to try the [preview release](https://github.com/Deep-Symmetry/beat-link-trigger/releases/tag/v7.4.0-SNAPSHOT) to let me know if it has resolved the problematic interaction between them, I would greatly appreciate it, and then we can close this issue if it works. With the help of Kevin in the Netherlands who was up researching while I slept (albeit not for enough hours), we have resolved the Windows build problem. Evidently GitHub upgraded their Actions infrastructure to PowerShell 7.3.0 which changed the way quoting works for arguments, and broke many people’s builds. Kevin found a setting we can use to restore the old behavior, which was weird, but which the script relied on, until we have time to switch to a newer version of the Wix installer toolkit. So the preview builds are back to supporting all platforms and on track for a future formal release.

**Comment by brunchboy** [comment]
If I don’t hear anything back soon, I’ll close this issue just hoping that my fix was successful.

**Comment by alstergee** [comment]
Sorry for the long delay. I can confirm the issue persists with BLT and
Vpns / tailscale being active

I'm sure it's got something to do with the handling of non-standard
Ethernet adapters, I've been having troubles with other programs like the
mars control software from Novastar not communicating with networked
Videowall processors when I have them on.

**Comment by brunchboy** [comment]
OK, there is probably nothing I can do about it in a cross-platform way, then. People will have to just not use that kind of network configuration, or use a separate Raspberry Pi or something to run BLT.

**Comment by brunchboy** [comment]
But just to be totally thorough, can you confirm you are using the latest 8.0.0 preview release, and share the log file with stack trace that still happens?

**Comment by alstergee** [comment]
Yeah sorry for delays I only get to test this when I'm running a show and I
don't own CDJs so it's hard to focus on like it's a lab environment heh

**Comment by brunchboy** [comment]
No need to apologize. I have invested many thousands of dollars in equipment to be able to further these projects, but I don’t expect most people to be able to do that! I appreciate your contributions whenever you do have time to make them. Even if I end up closing this issue again before you have time, I would still appreciate the information and we can reopen it then.

**Comment by brunchboy** [comment]
Having not heard anything back, I will close this agaiin.
